src/Processor/LocaleProcessor.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\PayPalPlugin\Processor;
+
+use Symfony\Component\Intl\Locales;
+
+final class LocaleProcessor implements LocaleProcessorInterface
+{
+    public function process(string $locale): string
+    {
+        if ($this->isValidLocale($locale)) {
+            return $locale;
+        }
+
+        $locales = Locales::getLocales();
+        $localeFound = (int) array_search($locale, $locales);
+        while (!$this->isValidLocale($locales[$localeFound]) && $localeFound < count($locales)) {
+            ++$localeFound;
+        }
+
+        return $locales[$localeFound];
```

Comment:
I'm still not 100% sure about this implementation :D indeed, it will not result in invalid locales (which is definitely an improvement), but it can result in very exotic default locales, like `fr_BE` for `fr`, `en_AE` for `en` and `de_AT` for `de` :/ 

I would still go with a hardcoded array of default locales. The alternative way would be an algorithm like:

1. Get the default locale from the array of the most common locales (like `en_US`, `fr_FR`, `de_DE` etc.)
2. If not, try to get the locale with the region being an uppercase version of locale (like it was before: `pl` -> `pl_PL`)
3. If not, get the locale with **this** implementation 

But IMO it's a little bit overcomplication for such a simple case :) 💃

Reply:
OK, I answer myself - still, the recommended configuration is to use the locale *with* region, so even if the service provides some exotic locale... it's only the default value :) I believe we will be able to fix it if it results in some problems 🖖